src/common/components/organisms/NogsGateButton.tsx

```
@@ -154,13 +136,62 @@ const NogsGateButton = (props: ButtonProps) => {
     };
   }, []);
 
-  const gatingSatisfied = hasNogs || userHoldEnoughSpace;
+  const isClient = typeof window !== "undefined";
+  const walletAddress = isClient ? (user?.wallet?.address as Address | undefined) : undefined;
+
+  const { data: spaceBalanceData } = isClient
+    ? useBalance({
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)